server/routes.go

```diff
@@ -1616,27 +1591,29 @@ func (s *Server) ChatHandler(c *gin.Context) {
 	}
 	msgs = filterThinkTags(msgs, m)
 
-	var harmonyMessageHandler *harmony.HarmonyMessageHandler
-	var harmonyToolParser *harmony.HarmonyToolCallAccumulator
-
-	useHarmony := shouldUseHarmony(m)
+	useHarmony := harmony.ShouldUseHarmony(m.Config.ModelFamily, m.Template)
 
 	processedTools := req.Tools
+	var functionNameMap *harmony.FunctionNameMap
+	var prefill string
+	// TODO(parthsareen/drifkin): renderer should be here for the template to allow prefill
+	// the tool mapping should live alongside the renderer and potentially move to the runner
 	if useHarmony {
-		harmonyMessageHandler = harmony.NewHarmonyMessageHandler()
+		functionNameMap = harmony.NewFunctionNameMap()
 		var lastMessage *api.Message
 		if len(msgs) > 0 {
 			lastMessage = &msgs[len(msgs)-1]
 		}
-		harmonyMessageHandler.HarmonyParser.AddImplicitStartOrPrefill(lastMessage)
-		harmonyToolParser = harmonyMessageHandler.CreateToolParser()
 
+		if lastMessage != nil {
+			prefill = harmony.Prefill(lastMessage.Role, lastMessage.Content, lastMessage.Thinking)
```

Comment:
how about just pass in the whole message and let harmony do whatever it wants with it? Like `harmony.Prefill(lastMessage)`

Reply:
got rid of a public prefill function since we can do the entire thing in harmony now